tests/compile-fail/function_calls/exported_symbol_abi_mismatch.rs

```diff
@@ -0,0 +1,22 @@
+// revisions: no_cache cache
+
+#[no_mangle]
+fn foo() {}
+
+fn main() {
+    #[cfg(cache)]
+    {
+        // `Instance` caching should not suppress ABI check.
+        extern "Rust" {
+            fn foo();
+        }
+        unsafe { foo() }
+    }
+    #[cfg_attr(cache, allow(clashing_extern_declarations))]
+    extern "C" {
+        fn foo();
+    }
+    unsafe { foo() }
```

Comment:
Could you add a version of this test where the function is imported via `extern "Rust"` but then the fn ptr is cast to `extern "C"` before being invoked?